x-pack/plugins/integration_assistant/server/graphs/ecs/model.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { prefixSamples } from '../../util/samples';
+import { ECS_EXAMPLE_ANSWER, ECS_FIELDS } from './constants';
+import { createPipeline } from './pipeline';
+import { mergeAndChunkSamples } from './chunk';
+import type { EcsMappingState } from '../../types';
+
+export function modelSubOutput(state: EcsMappingState): Partial<EcsMappingState> {
+  return {
+    lastExecutedChain: 'ModelSubOutput',
+    finalMapping: state.currentMapping,
```

Comment:
Will the `reducer` function get called at this point? Or is it in the `modelOutput`?

Reply:
The reducer is part of the state, so if you look at the entry in state.ts for finalMapping, you see that instead of setting a value by overriding it, we pass in the merge function again.
The X and Y that is always passed as an argument to these functions is the current value of the state and the update.